extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/grpc/GrpcTracingServerInterceptor.java

```diff
@@ -126,18 +122,20 @@ public void onHalfClose() {
 
         @Override
         public void onCancel() {
-            try {
+            Scope scope = spanContext.makeCurrent();
+            try (scope) {
```

Comment:
```suggestion
            try (Scope scope = spanContext.makeCurrent()) {
```
Just to prevent someone to use a closed scope after the try.